Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Controllers/MultiPositionController.cs

```diff
@@ -0,0 +1,1001 @@
+﻿// COPYRIGHT 2010, 2011, 2012, 2013 by the Open Rails project.
+// 
+// This file is part of Open Rails.
+// 
+// Open Rails is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// Open Rails is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License
+// along with Open Rails.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using Orts.Formats.Msts;
+using Orts.Parsers.Msts;
+
+namespace Orts.Simulation.RollingStocks.SubSystems.Controllers
+{
+    public class MultiPositionController
+    {
+        MSTSLocomotive Locomotive;
+        Simulator Simulator;
+
+        public List<Position> PositionsList = new List<Position>();
+
+        public bool Equipped = false;
+        public bool StateChanged = false;
+
+        public ControllerPosition controllerPosition = new ControllerPosition();
+        public ControllerBinding controllerBinding = new ControllerBinding();
+        protected float elapsedSecondsFromLastChange = 0;
+        protected bool checkNeutral = false;
+        protected bool noKeyPressed = true;
+        protected string currentPosition = "";
```

Comment:
This should be an enum. String comparison is less efficient

Reply:
Done